### ISSUE TITLE
MCFD Personal Request - Ministry User - Upload by Division

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#4569]
* As a MCFD Ministry User
* I want to be able to upload records by division
* so that they can upload records by division when needed

**Assumptions & Scope**	
This story arose out of a misinterpretation of requirements for MCFD personal requests. For most personal requests - IAO will be uploading records and tagging them by section through the scanning team. However; sometimes there will be personal requests that require traditional divisional upload, such as HR and employee files. 

In addition - the DDM team will have electronic (ICM) records that they will upload related to the requests themselves.

What is IN scope?
Ensuring Ministry personal requests can upload by division if needed

What is NOT in scope?

**Acceptance Criteria**	

**Scenario 1: MCFD Personal Requests - Upload Records with Divisions**
* GIVEN a Ministry Coordinator is on the records tab of a personal MCFD request
* AND divisions have been added
* WHEN they click on upload records
* THEN the records upload modal will appear

**Scenario 2: MCFD Personals - Upload Modal**
* GIVEN a MCFD Ministry user is on the upload record modal
* WHEN they click an added division from the tracking
* THEN that division will be selected

**Scenario 3: Add Files To Division**
* GIVEN a MCFD Ministry user has selected a division
* WHEN they have added files through drag and dropping
* OR through the file explorer
* THEN the upload button will become active

**Scenario 4: Add Files To Division - Upload**
* GIVEN a MCFD Ministry user has uploaded files to a selected division
* WHEN they click upload
* THEN the modal will close, and the files will be processed in the records tab

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by richard-aot** [comment]
Added the following ACs per our discussion in standup:
1. IAO scanning team member can only see sections in upload modal
2. IAO scanning team member can only edit section for the records they uploaded
3. CFD Ministry coordinators can only see selected divisions in upload modal
4. CFD Ministry coordinators can only edit division for the records they uploaded

@m-prodan @liseandtea Please review. Thanks.

**Comment by richard-aot** [comment]
@m-prodan @liseandtea Can you add CFD personal divisions here. We need to add them to DB through Admin tool or migration script. Thanks.

**Comment by nimya-aot** [comment]
@liseandtea Can you confirm If MSD personal requests should show has same divisions /sections as scanning team ? Got confused since @m-prodan mentioned some other divisions as the below screenshot? 


[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeTB3QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--38418aafe1d0ee180f8097b61e837875a115855a/image.png)

**Comment by nimya-aot** [comment]
@richard-aot  Could you please update the title to 'Add attachment' instead of 'Add scanned records' for the ministry user?
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBelV3QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1eba5eeaf2a47ce450a31531a9a9a01aefc74250/image.png)

**Comment by richard-aot** [comment]
@nimya-aot The title is updated for ministry user.

**Comment by nimya-aot** [comment]
@richard-aot intake user unable to see redaction app for the uploaded documents(scenario 5). Could you please check?
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMU0xQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--02192d90d7c8780a32222a81226845ffa01aceea/image.png)

**Comment by m-prodan** [comment]
As requested at stand up this morning by @abin-aot - I did some smoke tests of this feature, alongside admin app update. I'm able to add sections and divisions to personal requests only fairly easily. This works fairly intuitively. Will need to confirm with MCFD team how their other non adoptions/child protection related personals are handled, as right now the MCFD Personals requests no longer show the other divisions we have set up on the generals side, so we may just need to mimic what is in the generals side for personals too. This should make things easier for the MCFD Generals coordinator to deal with by not seeing all of the personals locations we will eventually add.

cc: @liseandtea
